server/etcdserver/server.go

```diff
@@ -1051,7 +1051,12 @@ func (s *EtcdServer) applySnapshot(ep *etcdProgress, apply *apply) {
 
 	// As backends and implementations like alarmsStore changed, we need
 	// to re-bootstrap Appliers.
-	s.uberApply = newUberApplier(s)
+	s.uberApply = s.NewUberApplier()
```

Comment:
This is even more fragile then before, can we remove some of those fields from server?

Reply:
It's not worse than what used to happen. 

We called restoreAlarms:
https://github.com/etcd-io/etcd/blob/a08d479463960948d5d9c3508623ade915c6e66f/server/etcdserver/server.go#L1003

That underneath were reconstructing the appliers and wrapping them:
https://github.com/etcd-io/etcd/blob/a08d479463960948d5d9c3508623ade915c6e66f/server/etcdserver/server.go#L2239-L2253

I'm afraid there are no fields to remove